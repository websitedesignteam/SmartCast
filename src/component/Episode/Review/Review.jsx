import React from 'react';
import { useState } from 'react';
import StarRating from '../StarRating/StarRating';
import styles from "./Review.module.scss";
import { isFormComplete } from "../../../utils/helper"; 

function Review({isLoadingReview, submitReview, ...props}) {
  	//states
    const [userReview, setUserReview] = useState({rating: 0, review: ""});

    const handleChange = (event) => {
		setUserReview({
			...userReview,
			[event.target.name]: event.target.value
		});
    };

    const setRating = (rating) => {
      	setUserReview({
			...userReview, 
			rating,
		});
    };

    const onClickSubmit = () => {
		submitReview(userReview);
    };

    return (
		<div className={styles["form"]}>
			<div className={styles["form-heading"]}>What did you think about this episode?</div>
			<div className={styles["form-input--rating"]}>
			<StarRating
				numberOfStars="5"
				currentRating="0"
				onClick={setRating}
				userReview={userReview}
			/>
			</div>
			<div className={styles["form-input--text"]}>
			<textarea
				name="review"
				rows="5"
				placeholder="Tell us your thoughts!"
				onChange={handleChange}
			/>
			</div>
			<div className={styles["form-bottom"]}>
			{!isLoadingReview
				? <button 
					className={styles["form-bottom--submit"]} 
					type="submit" 
					disabled={!isFormComplete(userReview)}
					onClick={onClickSubmit}
				>
					Submit Rating
				</button>
				: <div className="loaderTiny"></div>}
			</div>
		</div>
    );
}


export default Review;