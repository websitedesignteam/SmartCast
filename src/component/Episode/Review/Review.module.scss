@import "../../../styles/variables";

.form {
	border-radius: 1rem;
	box-shadow: $box-shadow;
	background-color: $light-gray;
	padding: 1.5rem;
	&-input {
		&--rating {
			cursor: pointer;
		}
		&--text {
			textarea {
				font-family: 'Cairo', sans-serif;
				width: 100%;
				-webkit-box-sizing: border-box; 
				-moz-box-sizing: border-box;    
				box-sizing: border-box; 
			}
		}
	}
	&-bottom {
		text-align: center;
		display: inline-flex;
		width: 100%;
		justify-content: center;
		margin-top: 0.5rem;
		&--submit {
			height: 2rem;
			border-radius: 1.5rem;
			font-size: 1rem;
			font-family: 'Cairo', sans-serif;
			padding: 0 1rem;
			background-color: $blue;
			&:hover:not([disabled]) {
				background-color: $blue-bright;
				outline: none;
			}
			&:focus {
				outline: $dark-gray;
			}
		}
	}
}

.loaderTiny {

}