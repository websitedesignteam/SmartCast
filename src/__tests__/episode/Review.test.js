import React from 'react';
import { shallow } from 'enzyme';
import Review from 'src/component/Episode/Review/Review';

test("should render the Review", () => {
    const submitReview = () => {}
    const props = {
        submitReview,

    }
    const component = shallow(<Review {...props} />);
    // expect(component.getElements()).toMatchSnapshot();
});