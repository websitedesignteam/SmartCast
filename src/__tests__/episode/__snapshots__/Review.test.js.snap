// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Review 1`] = `
Array [
  <div
    className="form"
  >
    <div
      className="form-heading"
    >
      What did you think about this episode?
    </div>
    <div
      className="form-input--rating"
    >
      <Rating
        currentRating="0"
        numberOfStars="5"
        onClick={[Function]}
        userReview={
          Object {
            "rating": 0,
            "review": "",
          }
        }
      />
    </div>
    <div
      className="form-input--text"
    >
      <textarea
        name="review"
        onChange={[Function]}
        placeholder="Tell us your thoughts!"
        rows="5"
      />
    </div>
    <div
      className="form-bottom"
    >
      <button
        className="form-bottom--submit"
        disabled={true}
        onClick={[Function]}
        type="submit"
      >
        Submit Rating
      </button>
    </div>
  </div>,
]
`;
